### PR TITLE
Add infra and ctest for code style formatting

### PR DESCRIPTION
**Pull request by drzaeus77**
This change adds a test for ensuring that changed files conform to some form of style guideline. Currently, that style is LLVM's.

### CODE REVIEW

**Comment by 4ast** [comment]
I'm missing something. Where is actual format spec? Once this pr lands than all non-llvm formated bits will be failing?

**Comment by drzaeus77** [comment]
The format spec is LLVM (not Google) defaults. I already started working on a few tweaks, and will take Vicent's .clang-format as well. The test should run only against .c, .h, and .cc files. Others will give incorrect results, especially python files.

And no, only if you touch a file will clang-format check it. That's why it's passing now, because nothing is in the changeset. Actually, it would be a good test, if you make a trivial change to some c/cc file and send a PR for it. It should flag it and give some helpful output.

**Comment by 4ast** [comment]
Then Vincent's clang-format should be part of the diff as well, otherwise interim other contributors will be confused?

**Comment by drzaeus77** [comment]
Yes yes, I'm working on it. Don't review this one any further until the next update.

**Comment by drzaeus77** [comment]
@4ast updated

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+BasedOnStyle: LLVM
```
Comment (4ast) [inline .clang-format]: should it be google?

Comment (drzaeus77) [inline .clang-format]: You're right, thinking about it a second time I hate their capitalization rules. Switching back to google.

**Comment by 4ast** [comment]
hope it's a step in the right direction. If it causes too much pain, we can reconsider.

**Pull request merged**